server/src/services/ito/transcribeStreamV2Handler.ts

```diff
@@ -0,0 +1,277 @@
+import { create } from '@bufbuild/protobuf'
+import { ConnectError } from '@connectrpc/connect'
+import {
+  ContextInfo,
+  ItoMode,
+  StreamConfig,
+  StreamConfigSchema,
+  TranscribeStreamRequest,
+  TranscriptionResponseSchema,
+} from '../../generated/ito_pb.js'
+import { getAsrProvider, getLlmProvider } from '../../clients/providerUtils.js'
+import { enhancePcm16 } from '../../utils/audio.js'
+import { DEFAULT_ADVANCED_SETTINGS } from '../../constants/generated-defaults.js'
+import { errorToProtobuf } from '../../clients/errors.js'
+import {
+  createUserPromptWithContext,
+  detectItoMode,
+  getPromptForMode,
+} from './helpers.js'
+import { ITO_MODE_SYSTEM_PROMPT } from './constants.js'
+import type { ItoContext } from './types.js'
+import { createWavHeader } from './audioUtils.js'
+
+interface ModeChangeRecord {
+  mode: ItoMode
+  timestamp: number
+}
+
+export class TranscribeStreamV2Handler {
```

Comment:
Now that we are handling midstream updates, this new handler has a few important changes over TranscribeV1 (left as a deprecated function we should support until all customers are on the new version)
* we merge context updates so that late arriving context overwrites pre existing context (this includes mode!)
* There is a mode change grace period at the end, where a user releasing keys may trigger a mode change. If it happens in a slim window at the end of the stream, we assume the user accidentally changed modes. Without this, a user with fn + ctl as intelligent mode might release ctl a fraction of a second before releasing fn, triggering a mode change to dictation at the very end.
* Audio and context are processed via the stream instead of headers